From eb1b930892da75135a1af6e377a3ebe27db76b85 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Fri, 13 Mar 2015 11:23:34 +0100
Subject: [PATCH 1/4] Allow overriding of pkg-config executable

Defined a PKG_CONFIG makefile variable and using it instead of the
hardcoded "pkg-config" value. This allows the use of a custom
pkg-config, e.g. "make PKG_CONFIG=i686-w64-mingw32-pkg-config", a quite
common convention adopted during cross compilation.
---
 lgi/Makefile   | 7 ++++---
 tests/Makefile | 9 +++++----
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/lgi/Makefile b/lgi/Makefile
index 8d3ad50..9ac8165 100644
--- a/lgi/Makefile
+++ b/lgi/Makefile
@@ -10,6 +10,7 @@ LUA_VERSION=5.1
 LUA_LIBDIR = $(PREFIX)/lib/lua/$(LUA_VERSION)
 LUA_SHAREDIR = $(PREFIX)/share/lua/$(LUA_VERSION)
 
+PKG_CONFIG = pkg-config
 GINAME = gobject-introspection-1.0
 PKGS = $(GINAME) gmodule-2.0 libffi
 VERSION_FILE = version.lua
@@ -37,14 +38,14 @@ ifndef COPTFLAGS
 CFLAGS = -Wall -Wextra -O2 -g
 endif
 endif
-ALL_CFLAGS = $(CCSHARED) $(COPTFLAGS) $(LUA_CFLAGS) $(shell pkg-config --cflags $(PKGS)) $(CFLAGS)
-LIBS += $(shell pkg-config --libs $(PKGS))
+ALL_CFLAGS = $(CCSHARED) $(COPTFLAGS) $(LUA_CFLAGS) $(shell $(PKG_CONFIG) --cflags $(PKGS)) $(CFLAGS)
+LIBS += $(shell $(PKG_CONFIG) --libs $(PKGS))
 ALL_LDFLAGS = $(LIBFLAG) $(LDFLAGS)
 DEPCHECK = .depcheck
 
 # Precondition check
 $(DEPCHECK) : Makefile
-	pkg-config --exists '$(GINAME) >= 0.10.8' --print-errors
+	$(PKG_CONFIG) --exists '$(GINAME) >= 0.10.8' --print-errors
 	touch $@
 
 .PHONY : all clean install
diff --git a/tests/Makefile b/tests/Makefile
index 322500e..24dda8f 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -25,20 +25,21 @@ endif
 
 PKGS = gio-2.0 cairo cairo-gobject gobject-introspection-1.0 gmodule-2.0 libffi
 LUA = lua
+PKG_CONFIG = pkg-config
 
 ifndef CFLAGS
 ifndef COPTFLAGS
 CFLAGS = -Wall -g
 endif
 endif
-ALL_CFLAGS = $(CCSHARED) $(COPTFLAGS) $(LUA_CFLAGS) $(shell pkg-config --cflags $(PKGS)) $(CFLAGS) -I .
-LIBS += $(shell pkg-config --libs $(PKGS))
+ALL_CFLAGS = $(CCSHARED) $(COPTFLAGS) $(LUA_CFLAGS) $(shell $(PKG_CONFIG) --cflags $(PKGS)) $(CFLAGS) -I .
+LIBS += $(shell $(PKG_CONFIG) --libs $(PKGS))
 ALL_LDFLAGS = $(LIBFLAG) $(LDFLAGS)
 DEPCHECK = .depcheck
 
 # Precondition check
 $(DEPCHECK) : Makefile
-	pkg-config --exists '$(PKGS) >= 0.10.8' --print-errors
+	$(PKG_CONFIG) --exists '$(PKGS) >= 0.10.8' --print-errors
 	touch $@
 
 REGRESS = $(PFX)regress$(EXT)
@@ -61,7 +62,7 @@ check : all
 $(REGRESS) : regress.o
 	$(CC) $(ALL_LDFLAGS) -o $@ regress.o $(LIBS)
 
-GIDATADIR = $(shell pkg-config --variable=gidatadir gobject-introspection-1.0)/tests
+GIDATADIR = $(shell $(PKG_CONFIG) --variable=gidatadir gobject-introspection-1.0)/tests
 
 regress.o : $(GIDATADIR)/regress.c $(GIDATADIR)/regress.h $(DEPCHECK)
 	$(CC) $(ALL_CFLAGS) -c -o $@ $<
-- 
2.3.2

